core/src/main/java/overflowdb/NodeDb.java

```diff
@@ -12,9 +12,9 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.locks.ReentrantLock;
```

Comment:
I don't think that we need an extra ReentrantLock: `synchronized` / the monitor does the job just fine.

Reply:
ah, you mean just `synchronized(this)`? yeah true, that will do the the same and is more concise :+1:

Reply:
actually, does it really? we want all write operations to be synchronized on the same lock, i.e. if one thread is setting an edge property we don't want to grow the adjacentNodes array at the same time in a different thread...

Reply:
as discussed: will push as synchronized because all synchronized(this) blocks lock on the same object monitor